constraint/pkg/apis/constraints/apis.go

```diff
@@ -45,3 +60,86 @@ func GetEnforcementAction(constraint *unstructured.Unstructured) (string, error)
 
 	return action, nil
 }
+
+func IsEnforcementActionScoped(action string) bool {
+	return action == EnforcementActionScoped
+}
+
+// GetEnforcementActionsForEP returns a map of enforcement actions for enforcement points.
+func GetEnforcementActionsForEP(constraint *unstructured.Unstructured, eps []string) (map[string]map[string]bool, error) {
+	actionsForEPs := make(map[string]map[string]bool)
+
+	for _, enforcementPoint := range eps {
+		actionsForEPs[enforcementPoint] = make(map[string]bool)
+	}
+
+	// Access the scopedEnforcementAction field
+	scopedActions, found, err := getNestedFieldAsArray(constraint.Object, "spec", "scopedEnforcementActions")
+	if err != nil {
+		return nil, fmt.Errorf("%w: invalid spec.enforcementActionPerEP", ErrInvalidConstraint)
+	}
+
+	// Return early if scopedEnforcementAction is not found
+	if !found {
+		return nil, fmt.Errorf("%w: spec.scopedEnforcementAction must be defined", ErrMissingRequiredField)
+	}
+
+	// Convert scopedActions to a slice of map[string]interface{}
+	scopedEnforcementActions, err := convertToMapSlice(scopedActions)
+	if err != nil {
+		return nil, fmt.Errorf("%w: spec.scopedEnforcementAction must be an array", ErrInvalidConstraint)
+	}
+
+	for _, scopedEnforcementAction := range scopedEnforcementActions {
+		scopedEA := &ScopedEnforcementAction{}
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(scopedEnforcementAction, scopedEA); err != nil {
+			return nil, err
+		}
+
+		// Iterate over enforcementPoints
+		for _, enforcementPoint := range scopedEA.EnforcementPoints {
+			if _, ok := actionsForEPs[enforcementPoint.Name]; !ok && enforcementPoint.Name != "*" {
+				continue
+			}
+			switch enforcementPoint.Name {
+			case "*":
+				for _, ep := range eps {
+					actionsForEPs[ep][scopedEA.Action] = true
```

Comment:
also need to make sure each scopedEA.Action is lower case to avoid case sensitivity issues.